moped-database/hasura-cluster

```diff
@@ -707,6 +291,57 @@ function show_banner() {
   echo "- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -";
 }
 
+function __check_export_complete() {
+  local file_path="$1"
+  local last_line=$(tail -n 3 "$file_path" | head -n 1)
+  
+  if [[ "$last_line" == *"PostgreSQL database export complete"* ]]; then
+    # echo "PostgreSQL database export is complete."
+    return 0
+  else
+    # echo "PostgreSQL database export is not complete."
+    return 1
+  fi
+}
+
+function replicate() {
+  source ./env;
+  start_only;
+  # Get the current date in YYYY-MM-DD format
+  current_date=$(date +"%Y-%m-%d");
+  snapshot_file="./snapshots/production-$current_date.sql";
+
+  # Check if the file already exists and the export is complete
+  if ! __check_export_complete "$snapshot_file"; then
+    echo "Replicating production database to local database... (~ 2-4 minutes)";
+    docker exec -it \
+    -e PGPASSWORD=$MOPED_RR_PASSWORD  \
+    moped-database-moped-pgsql-1 \
+    pg_dump \
+    -U $MOPED_RR_USERNAME \
+    -h $MOPED_RR_HOST \
+    $MOPED_RR_DB > $snapshot_file;
+    
+    # Check if the export is complete
+    if __check_export_complete "$snapshot_file"; then
+      echo "Replication successful.";
+    else
+      echo "Replication failed. The export is not complete.";
+      return 1;
```

Comment:
This should halt the program and by returning a non-zero return code, indicate an error.